### DISCUSSION TITLE
Example not working

### DISCUSSION THREAD

**Discussion by jtcookhome** [ExperienceLovelace/ha-floorplan#350] [General]
I setup a new dashboard, copy pasted teh multi_floor.yaml code  and I get this error.  I tried the simple and get the same error.  Its eems to be missing the views somehow

Your configuration is not valid: r: At path: views -- Expected an array value, but received: undefined

**Comment by OzGav** [comment]
I think there is something wrong with your yaml. If you can paste in your yaml making sure it is formatted exactly the same as in your editor (use the <> button)

**Comment by jtcookhome** [comment]
I copied / pasted direct from the example.  The ERROR being there is no "views"?

`
sidebar_title: Multi Floor
config:
  show_side_bar: false
  show_app_header: false
  config:
    image: /local/floorplan/examples/multi_floor/multi_floor.svg
    stylesheet: /local/floorplan/examples/multi_floor/multi_floor.css
     log_level: info
    console_log_level: info

    defaults:
      hover_action: hover-info
      tap_action: more-info

    startup_action:
      - service: floorplan.class_set
        service_data:
          element: ground_floor
          class: layer-visible
      - service: floorplan.class_set
        service_data:
          elements:
            - first_floor
          class: layer-hidden
      - service: floorplan.class_set
        service_data:
          element: ground_floor.button
          class: button-on
      - service: floorplan.class_set
        service_data:
          elements:
            - first_floor.button
          class: button-off

    rules:
      - element: ground_floor.button
        tap_action:
          - service: floorplan.class_set
            service_data:
              element: ground_floor
              class: layer-visible
          - service: floorplan.class_set
            service_data:
              elements:
                - first_floor
              class: layer-hidden
          - service: floorplan.class_set
            service_data:
              element: ground_floor.button
              class: button-on
          - service: floorplan.class_set
            service_data:
              elements:
                - first_floor.button
              class: button-off

      - element: first_floor.button
        tap_action:
          - service: floorplan.class_set
            service_data:
              element: first_floor
              class: layer-visible
          - service: floorplan.class_set
            service_data:
              elements:
                - ground_floor
              class: layer-hidden
          - service: floorplan.class_set
            service_data:
              element: first_floor.button
              class: button-on
          - service: floorplan.class_set
            service_data:
              elements:
                - ground_floor.button
              class: button-off

      - entities:
          - binary_sensor.garage
          - binary_sensor.activity
        state_action:
          service: floorplan.class_set
          service_data: '${ entity.state === "on" ? "binary-sensor-on" : "binary-sensor-off"}'
`

**Reply by jtcookhome** [reply]
i'm not sure why the firs part before defaults" isn't formatting as code

**Reply by jtcookhome** [reply]
is there a different version that the yaml is formatted different? all of the example yaml files do not have the "views"

I installed ha-floorplan through HACS

**Reply by exetico** [reply]
Hi, I'm glad that you figured it out. Have you checked this video?
https://youtu.be/7AU5i-WUUxw

In general, there's multiple ways to work with YAML. Why do you need a different version of the examples? It sounds like you can put the missing parts together.

Let me know if there's something we should consider, to improve the first setup.

**Reply by OzGav** [reply]
The yaml should start something like. 

```

panel: true
cards:    
  - type: 'custom:floorplan-card'    
    full_height: true        
    config:
      log_level: debug
      stylesheet:
        location: /local/floorplan/floorplan-remote.css
        cache: true
        image:
```

**Reply by jtcookhome** [reply]
@exetico 
I haven't figured it out.  I tried different examples because they do not work.  They appear to be missing some starting lines of code.

If the examples are to help someone get started, they should be complete to copy/paste.

**Reply by OzGav** [reply]
You need to read all the docs. @exetico did an excellent write up here https://community.home-assistant.io/t/floorplan-now-available-as-a-lovelace-card/115489/323?u=ozgav

The examples are there to show you what is possible with the key parts of the yaml shown.

**Comment by exetico** [comment]
@jtcookhome  You must have missed something. A Home Assistant YAML config needs a view to work, and even more than that. That's shown a few times in the videos I've created, so I'd suggest you to give them a re-watch.

Below is a long post with random examples and inputs. I hope that helps? 👌 You'd need to understand that Home Assistant has a few ways to edit YAML code nowadays. So it will be a bit tricky for new users to understand the structure. I fully understand that, and that's why I've created the videos.

Here's a random example from my playground:

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/9f921a71-0184-441d-8c36-e6cfecebf783)

Take a look at the structure. Here's the "title", the "views" and the "cards".

_I bet there's a few odd things or similar, as this is a testing case. But that's a full example of a YAML code, which are edited in the "Raw configuration editor":_

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/025ed13e-1e10-4267-9907-d4ecaadd4aa7)

A YAML Lovelave config edited in the Raw Configuration Editor can contains multiple things.

<hr>

Here's another random Raw Configuration Editor with a view, and come cards. That's basic Lovelace rules to set up it like that.

I understand, as a totally new "YAML editor", that it can be a bit hard to get your head around everything, but I'd suggest you to watch a few videos with the first initial steps of how to format and configure Lovelace in Home Assistant.

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/ffac0250-bfd3-452d-a93d-30493ca027ce)

If you're using the graphical "Edit Dashboard" option in Home Assistant, you're able to play around with other cards (ha-floorplan does NOT support that by default), but that will help you understand the setup.

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/89dc0599-6dd9-42fa-b8e4-224b69b1ae0d)

I hope this helps a bit. Below is another example, if you've edited the Lovelave layout in the GUI editor.

**First example with a random view**:

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/6c1e0cd3-b333-4b7c-a684-8eeb484eb964)

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/19903fd5-2315-4ff7-a607-73878edc5d7d)

**Secound example with ha-floorplan**:

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/3e88b1f3-903f-4bc8-ac02-8ee9b824d67a)

And format the code like this:

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/f80fda96-238e-4f96-b971-780648849014)

**After** you've added ha-floorplan the first time, you can also use the `Edit` button in the bottom of the GUI, to edit the ha-floorplan code itself:

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/f96e763a-e116-46db-8221-99dbe0f2ada9)

Here you'll find the structure matching our example, as we expect our users to have at least a bit of understanding of the generel structure. It's also cool if our users needs help - that's why the videos can help you through the initial steps, in a fresh home-assistant setup.

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/1666e9ff-f95f-4f83-9774-2aaf0c12e0cc)


I'l also like to point out that this is **very clear'* in my video, so maybe you've misunderstood something in the video, somehow? :)

Please take a look at this:
https://youtu.be/5ue9J4WG2a8?t=690

If you use the GUI for everything except ha-floorplan, you can also add the view, press the "Add card" and select "Manuel"

Here the example will work, too:

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/b1ab1871-4fe0-493e-bb1d-9b8fad9af73b)

![image](https://github.com/ExperienceLovelace/ha-floorplan/assets/3549445/b4a0a120-5c39-4dad-b19c-75e65772204a)
_I've just used some random code from another case._

Happy floorplanning 🎉 !

**Reply by jtcookhome** [reply]
@exetico, I orginally had the simple floorplan working then saw the other examples and wanted to try the multifloor as that fits my application.

thank you for the response, I will watch the video you shared.

**Reply by exetico** [reply]
You're welcome 😄!

If so, you shoud now have plenty of details on how to continue with the screenshots provided.

**Reply by mtojay** [reply]
holy moly. @exetico i dont really have anything to add to this thread, but i just wanted to let you know that this is some a grade support. that is amazing.
